packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -22,10 +24,11 @@ const SIMPLE_SERVER_FILE_PATH = join(__dirname, "simple-server", "bundled_server
 export async function checkBuildConditions(cwd: string): Promise<void> {
   // dynamically load Angular so this can be used in an NPX context
   const { NodeJsAsyncHost }: typeof import("@angular-devkit/core/node") = await import(
-    `${cwd}/node_modules/@angular-devkit/core/node/index.js`
+    require.resolve("@angular-devkit/core/node/index.js", { paths: [cwd] })
```

Comment:
Why do we need the paths:[cwd]?

If we remove this I think it will let us support yarn pnp once the Angular build issue is fixed

Reply:
@jamesdaniels James could you explain why only with cwd, npx would be supported? And if it's safe to remove this so that we can support yarn pnp?

Reply:
So paths is a hint telling NPM to resolve in the specified directories. This is important incase this script is called via NPX, globally installed, or if the developer if using a version manager like Volta—as we'd want these dependencies to be resolved specifically for the project at hand. 

Not a big deal for the buildpack but this guards the web frameworks migration, integration testing, and emulation paths in the future.

Reply:
I think we run into a problem with yarn plugnplay if we restrict npm to only checking in cwd. Yarn pnp (https://yarnpkg.com/features/pnp) doesn't use node_modules but uses a cache instead where it installs the dependencies.

For example some app using yarn pnp might have the @angular-devkit/core module resolve to ../../.yarn/cache/@angular-devkit/core. If we keep "paths: [cwd]" then it will never find this directory.

The location of this cache directory is also optionally configurable via a field in .yarnrc.yml so its also not as simple as just adding another path that we know will contain this .yarn cache.

Going forward if we want to support yarn pnp we will need to either drop this cwd or add in some custom logic to find the .yarn dir to add into paths as well. I would lean towards dropping the cwd here for now and revisiting this if it becomes a bigger issue in the future, I'm not the biggest fan of adding a good amount of custom logic to support yarn pnp here specifically. Open to other thoughts or ideas as well. wdyt @jamesdaniels?

Reply:
Shouldn't corepack handle that for us? Are we enabling it in the buildpack if package manager != npm

Reply:
Let's get these changes in now imo. We should probably get corepack to handle this for us in the long term but we probably want to have pnpm working for now.